scripts/update/ItemUpdater.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ *
+ */
+
+namespace oat\taoQtiItem\scripts\update;
+
+use oat\taoQtiItem\model\qti\ParserFactory;
+use \RecursiveIteratorIterator;
+use \RecursiveDirectoryIterator;
+
+abstract class ItemUpdater
+{
+    protected $itemPath     = '';
+    protected $checkedFiles = array();
+
+    /**
+     * Init the item updater with the item directory path
+     * 
+     * @param string $itemRootPath
+     * @throws \common_Exception
+     */
+    public function __construct($itemRootPath)
+    {
+        if(file_exists($itemRootPath)){
+            $this->itemPath = $itemRootPath;
+        }else{
+            throw new \common_Exception('the given item root path does not exist');
+        }
+    }
+
+    /**
+     * Update all the item files found within the $itemRootPath
+     * @param boolean $changeItemContent - tells if the item files will be written with the updated content or not
+     * @return array of modified item instances
+     */
+    public function update($changeItemContent = false)
+    {
+        $returnValue = array();
+        $objects     = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->itemPath), RecursiveIteratorIterator::SELF_FIRST);
+
+        foreach ($objects as $itemFile => $cursor) {
+
+            $this->checkedFiles[$itemFile] = false;
```

Comment:
I guess that this line should be wrapped in `is_file()` function because in this array falls not only files, but directories and such elements like `.` and `..`